tensorly/decomposition/tests/test_tucker.py

```diff
@@ -93,14 +90,12 @@ def test_tucker(monkeypatch):
         assert_equal(
             factors[i].shape,
             (tensor.shape[i], ranks[i]),
-            err_msg="factors[{}].shape={}, expected {}".format(
-                i, factors[i].shape, (tensor.shape[i], ranks[i])
-            ),
+            err_msg=f"{factors[i].shape = }, expected {(tensor.shape[i], ranks[i])}"
         )
         assert_equal(
             tl.shape(core)[i],
             rank,
-            err_msg="Core.shape[{}]={}, " "expected {}".format(i, core.shape[i], rank),
+            err_msg=f"{core.shape[i] = }, expected {rank}"
```

Comment:
Changed to support Python 3.7

```suggestion
            err_msg=f"core.shape[i] = {core.shape[i]}, expected {rank}"
```